File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.py
class NumArray(object):

    def __init__(self, nums):
        self.answer = nums
        self.length = len(nums)
        for i in range(1,self.length):
            self.answer[i] += self.answer[i-1]

    def sumRange(self, left, right):
        if left!=0:
            return self.answer[right]-self.answer[left-1]
        else:
            return self.answer[right]
            
        
        


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)

File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.cpp
class NumArray {
public:
    vector<int> answer;
    NumArray(vector<int>& nums) {
        
        answer = nums;
        for(int i=1;i<nums.size();i++)
        {
            answer[i] += answer[i-1];
        }
        
    }
    
    int sumRange(int left, int right) {
        if(left != 0)
        {
            return answer[right] - answer[left-1];
        }
        else
            return answer[right];
        
    }
};

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray* obj = new NumArray(nums);
 * int param_1 = obj->sumRange(left,right);
 */